docs/docs/understanding/agent/basic_agent.md

```diff
@@ -0,0 +1,116 @@
+
+![agent flow](./agent_flow.png)
+
+## Getting started
+
+You can find all of this code in [the tutorial repo](https://github.com/run-llama/python-agents-tutorial).
+
+To avoid conflicts and keep things clean, we'll start a new Python virtual environment. You can use any virtual environment manager, but we'll use `poetry` here:
+
+```bash
+poetry init
+poetry shell
+```
+
+And then we'll install the LlamaIndex library and some other dependencies that will come in handy:
+
+```bash
+pip install llama-index python-dotenv
+```
+
+If any of this gives you trouble, check out our more detailed [installation guide](../getting_started/installation/).
+
+## OpenAI Key
+
+Our agent will be powered by OpenAI's `GPT-3.5-Turbo` LLM, so you'll need an [API key](https://platform.openai.com/). Once you have your key, you can put it in a `.env` file in the root of your project:
+
+```bash
+OPENAI_API_KEY=sk-proj-xxxx
+```
+
+If you don't want to use OpenAI, we'll show you how to use other models later.
+
+## Bring in dependencies
+
+We'll start by importing the components of LlamaIndex we need, as well as loading the environment variables from our `.env` file:
+
+```python
+from dotenv import load_dotenv
+
+load_dotenv()
+from llama_index.core.agent import ReActAgent
+from llama_index.llms.openai import OpenAI
+from llama_index.core.tools import FunctionTool
+```
+
+## Create basic tools
+
+For this simple example we'll be creating two tools: one that knows how to multiply numbers together, and one that knows how to add them.
+
+```python
+def multiply(a: float, b: float) -> float:
+    """Multiply two numbers and returns the product"""
+    return a * b
+
+
+multiply_tool = FunctionTool.from_defaults(fn=multiply)
+
+
+def add(a: float, b: float) -> float:
+    """Add two numbers and returns the sum"""
+    return a + b
+
+
+add_tool = FunctionTool.from_defaults(fn=add)
+```
+
+As you can see, these are regular vanilla Python functions. The docstring comments provide metadata to the agent about what the tool does. We then create `FunctionTool` objects from these functions, which wrap them in a way that the agent can understand.
+
+## Initialize the LLM
+
+`GPT-3.5-Turbo` is going to be doing the work today:
+
+```python
+llm = OpenAI(model="gpt-3.5-turbo", temperature=0)
+```
+
+You could also pick another popular model accessible via API, such as those from [Mistral](../examples/llm/mistralai/), [Claude from Anthropic](../examples/llm/anthropic/) or [Gemini from Google](../examples/llm/gemini/).
```

Comment:
we could mention open-source models here (but open-source models also suck at this right now, so maybe its best left out?)